Staircase.js

```diff
@@ -1,37 +1,61 @@
+/*
+TODO:
+    * Improve documentation?
+*/
 function Staircase(stairs) {
   this.stairs = {};
   for (var i in stairs) {
     this.stairs[i] = stairs[i];
-    this.stairs[i].firstVal = stairs[i].firstVal;
-    this.stairs[i].down = stairs[i].down;
-    this.stairs[i].up = stairs[i].up || 1;
-    this.stairs[i].factor = stairs[i].factor;
-    this.stairs[i].direction = stairs[i].direction;
-    this.stairs[i].limits = stairs[i].limits;
-    this.stairs[i].operation = stairs[i].operation;
-    this.stairs[i].wait = stairs[i].wait || false;
-    this.stairs[i].sameStairMax = stairs[i].sameStairMax;
     this.stairs[i].name = i;
-    this.stairs[i].val = stairs[i].val || [this.stairs[i].firstVal];
-    this.stairs[i].active = stairs[i].active || false;
-    this.stairs[i].limitReached = stairs[i].limitReached || false;
-    this.stairs[i].successiveGood = stairs[i].successiveGood || 0;
+    // Check minimum requirements: first value and stepSize
+    if (typeof stairs[i].firstVal=="undefined" && typeof stairs[i].limits != "object")
+        throw new Error("No firstVal specified for "+i);
+    else if(stairs[i].hasOwnProperty("firstVal"))
+        this.stairs[i].firstVal = stairs[i].firstVal;
+    else
+        this.stairs[i].firstVal = stairs[i].limits[0]+((stairs[i].limits[1]-stairs[i].limits[0])/2); // use the midpoint between limits if limits are specified and firstVal is not
+    if (typeof stairs[i].stepSize=="undefined")
+        throw new Error("No stepSize specified for "+i);
+    else
+        this.stairs[i].stepSize = stairs[i].stepSize;
+    // Everything else we can derive sensible defaults for on the basis of
+    // first value and step size if not specified
+    // NOTE: easier is always 'up', even if it is a numerical decrease in val
+    this.stairs[i].down = stairs[i].down || 1; // N-down in 'wait' mode, LEGACY: scales stepSize when doing down-steps (getting harder) in 'no-wait' mode
+    this.stairs[i].up = stairs[i].up || 1; // N-up in 'wait' mode, LEGACY: scale stepSize when doing up-steps (getting easier) in 'no-wait' mode
+    this.stairs[i].downUpRatio = stairs[i].downUpRatio || 1; // delta-Down/delta-Up (with the x-up,y-down rule controls the convergence point). Not implemented for 'multiply' mode
+    this.stairs[i].direction = stairs[i].direction || 1; // -1: lower val is easier | 1: lower val is harder
+    this.stairs[i].reversalLimit = stairs[i].reversalLimit || 0; // Maximum reversals before settling on the final value. 0: infinite
+    this.stairs[i].limits = stairs[i].limits || false;
+    this.stairs[i].operation = stairs[i].operation || 'add'; // Modes: 'add' | 'multiply'
+    this.stairs[i].wait = stairs[i].wait || (this.stairs[i].down!=this.stairs[i].up); // By default we wait if up and down are different
```

Comment:
Fair point. I use a few questionable defaults like this one in order to try to avoid breaking code which depeneded upon your no/wait stuff. 

You being the author you're more entitled to break things :)